todoapp/todos/models.py

```
@@ -15,4 +16,20 @@ class Todo(models.Model):
+    date_created = models.DateTimeField(default=now)
+    date_completed = models.DateTimeField(null=True, blank=True)
+
+    def save(self, *args, **kwargs):
+        if self.done and self.date_completed is None:
+            self.date_completed = now()
+        elif not self.done:
+            self.date_completed = None
+        super().save(*args, **kwargs)
+
+    def __str__(self):
+        return self.name
```

Ruff (0.12.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: Magic method should come before `save` method

(DJ012)